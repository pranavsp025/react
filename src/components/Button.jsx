import React from "react";
const Button = ({text,handleSubmit}) => {
    return(

        <button className="loginbutton" type="submit" data-testid="button-test-id" >{text}</button>
    )
}
export default Button;