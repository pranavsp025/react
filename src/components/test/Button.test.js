import React from "react"
import { render,fireEvent } from "@testing-library/react"
import Button from '../Button';

describe("Check if Button works properly", () =>{
    test("Check if Button rendered correctly",()=>{
        const {getByTestId} = render(<Button />);

        const element = getByTestId('button-test-id');


        expect(element).toBeTruthy()
    })
    
    test("Check if text displayed properly",()=>{
        const text="Click";
        
        const { getByText } = render(<Button text={text}/>)

        getByText(text);
    })

    test("Check if onClick is triggerred", () => {
        const onClick = jest.fn();

        const {getByTestId} = render(<Button onClick={onClick}/>);
        const element = getByTestId('button-test-id');

        fireEvent.click(element);

        expect(onClick).not.toHaveBeenCalledTimes(1)
    })

    test("Check is snapshots is matched properly",() => {
        const onClick = jest.fn();
        const text = "Click";

        const { asFragment } = render(<Button text={text} onClick={onClick}/>);

        expect(asFragment()).toMatchSnapshot();
    })
})