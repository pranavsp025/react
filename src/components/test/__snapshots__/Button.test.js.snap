// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Button works properly Check is snapshots is matched properly 1`] = `
<DocumentFragment>
  <button
    class="loginbutton"
    data-testid="button-test-id"
    type="submit"
  >
    Click
  </button>
</DocumentFragment>
`;
